### ISSUE TITLE
Proposal for Network Change Observable

### ISSUE THREAD

**Issue by Laimiux** [ReactiveX/RxAndroid#128]
I think observing network changes would be useful in many projects, just not sure if this belongs in RxAndroid or a separate module. What is your opinion?

### Here is a working code example

```java
public final class NetworkChangeReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Network.changed(context);
    }
}
```

```java
public class Network {
    public static enum State {
        NOT_CONNECTED,
        MOBILE,
        WIFI
    }

    private static BehaviorSubject<State> behaviorSubject = BehaviorSubject.create();

    private Network() {
        // No instances
    }

    static void changed(Context context) {
        behaviorSubject.onNext(getConnectivityStatus(context));
    }

    /**
     * Helper function that returns the connectivity state
     *
     * @param context Context
     * @return Connectivity State
     */
    public static State getConnectivityStatus(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (null != activeNetwork) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI)
                return State.WIFI;

            if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE)
                return State.MOBILE;
        }
        return State.NOT_CONNECTED;
    }

    /**
     * @warn Add documentation
     */
    public static Observable<State> observe(Context context) {
        changed(context);
        return observe();
    }

    /**
     * @warn Add documentation
     */
    public static Observable<State> observe() {
        return behaviorSubject.asObservable();
    }
}
```

**Comment by ecokeley** [comment]
Nice use of package scope =)

**Comment by SaifurRahmanMohsin** [comment]
Hey, do you got an example for this?

**Comment by SaifurRahmanMohsin** [comment]
Got an example for this? Like a minimal app which shows Toast messages whenever network connectivity changes. I’m trying to implement this in my own app and a sample would be of great help. Thanks!

**Comment by Laimiux** [comment]
Here is a link to sample:
https://github.com/Laimiux/rxnetwork-android

As I have looked over RxAndroid code more, I realized there is already a better way to stream connectivity changes. Since, I don't think we need to receive connectivity updates while app is not running, we can instead leverage LocalBroadcastReceiver(ContentObservable.fromBroadcast). So, no need to declare broadcast receiver in AndroidManifest anymore.
